ph-stx-engine/src/main/java/net/sf/joost/stx/function/IndexOf.java

```diff
@@ -72,11 +72,17 @@ public Value evaluate (final Context context, final int top, final AbstractTree
     Value seq = args.m_aLeft.evaluate (context, top);
     final Value item = args.m_aRight.evaluate (context, top);
 
-    if (seq.type == Value.EMPTY)
+    if (seq.type() == Value.EMPTY)
       return seq;
 
     final AbstractTree tSeq = new ValueTree (seq);
-    item.next = null;
+
+    // TODO Guard around the supposed "Immutable" types in Value, but allow current behaviour for others ...
+    // This seems like thoroughly dangerous behaviour though. We're modifying a caller value in a function
+    // that does not indicate modification of this parameter.
+    if(item.isMutableValue()) {
+      item.next(null);
```

Comment:
No, I didn't know. Which worries me a bit. Especially since we're modifying a parameter that callers would likely expect to remain untouched. Can completely remove it, or put an error in if a parameter is passed that is a sequence?